src/components/course-settings/roster.cjsx

```diff
@@ -14,10 +14,27 @@ module.exports = React.createClass
   propTypes:
     courseId: React.PropTypes.string.isRequired
 
+  updateCourseName:(courseId)->
+    RosterActions.save(@props.courseId)
+    console.log('waka kinda working');
+
+  changeCourseName:->
+    course = CourseStore.get(@props.courseId)
+    <BS.Popover tilte={'Change Course Name'} {...@props}>
+      <form onSubmit={@updateCourseName} >
+        edit <input className='courseTitle' ></input>
+         <input type="submit" className="submit" value="rename" onSubmit={@updateCourseName}/>
+      </form>
+    </BS.Popover>
+
   render: ->
     course = CourseStore.get(@props.courseId)
     tabs = _.map course.periods, (period, index) =>
       <BS.TabPane key={period.id}, eventKey={index} tab={period.name}>
+        <BS.OverlayTrigger rootClose={true} trigger='click' placement='right'
+          overlay={@changeCourseName()}>
```

Comment:
I think this could just be a variable instead of calling a method. The `change` in the name of the method is a bit confusing: does it mean calling `changeCourseName` will change the course name? By convention in tutor-js, when a method generates DOM Elements the method is prefixed with `render...` but I think in the case of a popover you could just create a variable inside `render: ->` named `courseNamePopover`